src/clients/BridgeClient.ts

```diff
@@ -61,6 +63,30 @@ export class BridgeClient implements IClient {
                 this._logger.trace(TelemetryEvent.BridgeClient_GetVersionSuccess);
                 return bridgeVersion;
             };
+            const codesignAsyncFn = async (): Promise<string> => {
+                const args: string[] = [ `-s`, `-`, this._executablePath ];
+                return await this._commandRunner.runAsync(
+                    this.codesignPath,
+                    args,
+                    null, // currentWorkingDirectory
+                    null  // customEnvironmentVariables
+                );
+            };
+
+            if (process.platform === `darwin` 
+                && process.arch === `arm64` 
+                && await fileSystem.existsAsync(this.codesignPath)) {
+                try {
+                    const output = await RetryUtility.retryAsync<string>(codesignAsyncFn, 1, 100);
```

Comment:
Tested this with M1 Mac - after code sign it works without any issue - reference here - https://github.com/dotnet/runtime/issues/49091#issuecomment-797029172

Reply:
NIce, quick question please: this all happens at runtime? or Binary will be codesign by this.

Reply:
@Tatsinnit this code sign tool/library is specify to m1 Mac and code sign can be done only from m1 Mac. I searched for doing it from different OS but didn't find any options. So only way is to sign them during runtime. Please let me know or if you find further information about it. Thanks.

Reply:
I know very little about Macs, but a google search turned up [this](https://localazy.com/blog/how-to-automatically-sign-macos-apps-using-github-actions), which suggests that it can be done on the `macos` GitHub runners. I'd expect it to be simpler in this case because it's not being signed with an identity/certificate.

Reply:
@peterbom I agree and thought about it earlier as well. We need to move all the internal azure pipelines where all the binaries are generated (dotnet publish), then code sign for windows happens with MS certificates and uploaded to storage account. This all needs to be moved to GitHub actions plus GitHub actions doesn't support code sign for windows with MS certs. I can sync with you offline to make this more clear. Please let me know. Thanks.

Reply:
Ah, yes: we're generating the binaries in our internal ADO pipeline (which we can't move to GH yet).

And that pipeline doesn't support MacOS runners - is that the problem?

Reply:
@peterbom I want to say yes, but I want to explore more before answering. Azure pipelines does support mac os runners but the internal pipelines are in different org and I don't know if their agent pool would support mac os. I will do some research and get back to you. Thanks.